src/mongocrypt-ctx-encrypt.c

```diff
@@ -2092,10 +2092,6 @@ mongocrypt_ctx_explicit_encrypt_init (mongocrypt_ctx_t *ctx,
       }
       /* algorithm is FLE 1 only. */
       if (ctx->opts.algorithm != MONGOCRYPT_ENCRYPTION_ALGORITHM_NONE) {
-         if (ctx->opts.index_type.set) {
-            return _mongocrypt_ctx_fail_w_msg (
-               ctx, "cannot set both algorithm and index type");
-         }
```

Comment:
Correct. After that change, there should be no code path in which a user can specify both the algorithm and the index_type, since they both pass through the same function that already checks that condition.

Reply:
`_mongocrypt_ctx_init` seems slightly preferable to error for any context type.